cmd/admin-handlers.go

```diff
@@ -400,8 +403,71 @@ func (adminAPI adminAPIHandlers) ClearLocksHandler(w http.ResponseWriter, r *htt
 	writeSuccessResponseJSON(w, jsonBytes)
 }
 
-// validateHealQueryParams - Validates query params for heal list management API.
-func validateHealQueryParams(vars url.Values) (string, string, string, string, int, APIErrorCode) {
+// ListUploadsHealHandler - similar to listObjectsHealHandler
+// ListUploadsHealHandler -
+// GET /?heal&bucket=mybucket&prefix=myprefix&key-marker=mymarker&upload-id-marker=myuploadid&delimiter=mydelimiter&max-uploads=1000
+// - bucket is mandatory query parameter
+// - rest are optional query parameters
+// List upto maxKey objects that need healing in a given bucket matching the given prefix.
+func (adminAPI adminAPIHandlers) ListUploadsHealHandler(w http.ResponseWriter, r *http.Request) {
+	// Get object layer instance.
+	objLayer := newObjectLayerFn()
+	if objLayer == nil {
+		writeErrorResponse(w, ErrServerNotInitialized, r.URL)
+		return
+	}
+
+	// Validate request signature.
+	adminAPIErr := checkRequestAuthType(r, "", "", "")
+	if adminAPIErr != ErrNone {
+		writeErrorResponse(w, adminAPIErr, r.URL)
+		return
+	}
+
+	// Validate query params.
+	vars := r.URL.Query()
+	bucket := vars.Get(string(mgmtBucket))
+	prefix, keyMarker, uploadIDMarker, delimiter, maxUploads, _ := getBucketMultipartResources(r.URL.Query())
+	if maxUploads < 0 || maxUploads > maxUploadsList {
+		writeErrorResponse(w, ErrInvalidMaxUploads, r.URL)
+		return
+	}
+
+	// Check if bucket name is valid.
+	if !IsValidBucketName(bucket) {
+		writeErrorResponse(w, ErrInvalidBucketName, r.URL)
+		return
+	}
+
+	// Check if prefix is valid.
+	if !IsValidObjectPrefix(prefix) {
+		writeErrorResponse(w, ErrInvalidObjectName, r.URL)
+		return
+	}
+
+	// Marker not common with prefix is not supported.
+	if keyMarker != "" {
+		if !hasPrefix(keyMarker, prefix) {
+			writeErrorResponse(w, ErrNotImplemented, r.URL)
+			return
+		}
+	}
+
+	// Get the list objects to be healed.
+	listMultipartInfos, err := objLayer.ListUploadsHeal(bucket, prefix,
+		keyMarker, uploadIDMarker, delimiter, maxUploads)
+	if err != nil {
+		writeErrorResponse(w, toAPIErrorCode(err), r.URL)
+		return
+	}
+
+	listResponse := generateListMultipartUploadsResponse(bucket, listMultipartInfos)
+	// Write success response.
+	writeSuccessResponseXML(w, encodeResponse(listResponse))
+}
+
+// validateListObjectsHealQuery - Validates query params for heal objects list management API.
+func validateListObjectsHealQuery(vars url.Values) (string, string, string, string, int, APIErrorCode) {
```

Comment:
extractListObjectsHealQuery is better than validateListObjectsHealQuery since it is also returning values.